crates/core/args.rs

```diff
@@ -648,15 +651,16 @@ impl ArgMatches {
                     "~".repeat(79),
                     pcre_err,
                 )))
-            },
+            }
 
-            _ => Err(From::from(format!("Engine '{}' not found. Refer to the \
-                                         help to know supported values.",
-                                        engine)))
+            _ => Err(From::from(format!(
+                "Engine '{}' not found. Refer to the help to know supported \
+                 values.",
+                engine
+            ))),
```

Comment:
Could you please squash this commit back into the pertinent commits? I try to keep ripgrep's commit history clean, readable and linear, with tests passing on all commits. I don't always succeed, but I'd like to try. There isn't too much code here, so I think it's not too big of a pain to split these changes and squash them back into appropriate commits.

Although, I guess now that I say this, the commit hygiene in this PR is a bit off. It probably would have been better to start by refactoring how matchers are built and have that in one commit. And then a separate commit that actually adds the new `--engine` flag, which would include all relevant changes (including completions and docs and tests).

Reply:
Ok yes thanks, I see what you meant by "commit hygiene", seeing modifications that way does make more sense.

Seeing all the changes needed to be done, is it OK if we just close/throw away this PR and make a new one ? Thanks.

Reply:
Maybe throw it away, do a first PR for the CI build, then do a second PR for the refactor+engine option ? 

I'm actually impressed by the speed of your answers ! I will wait for your confirmation before doing that though :)

Reply:
Up to you. I don't have a strong opinion. I'm happy to have you force push stuff here, but if a separate PR is easier for you, then go for it.

As for the CI fix, sure, a separate PR seems fine for that. Thanks. :-)

Reply:
> I'm actually impressed by the speed of your answers 

Thanks. I'm not always this fast though. You just happened to catch me at a good time. :)